drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,296 @@
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL9_XL, tmp);
+    res += i2c_write_reg(dev->i2c, dev->acc_address
+                         LSM6DS3_REG_CTRL1_XL, acc_scale);
+    if (res != 3)
+    {
+        DEBUG("[ FAIL ]\n");
+    }
+    DEBUG("[ OK ]\n");
+
+	  /* configure accelerometer data ready pin */
+	  gpio_init_in(acc_pin, GPIO_NOPULL);
+
+	  /* configure temperature */
+    DEBUG("LSM6DS3: configure temperature ... ");
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL4_C, LSM6DS3_CTRL4_C_TEMP_EN);
+    if (res != 4)
+    {
+        DEBUG("[ FAIL ]\n");
+    }
+    DEBUG("[ OK ]\n");
+
+    /* configure gyroscope */
+    DEBUG("LSM6DS3: configure gyroscope ... ");
+    tmp = (LSM6DS3_CTRL10_G_X_EN
+          | LSM6DS3_CTRL10_G_Y_EN
+          | LSM6DS3_CTRL10_G_Z_EN);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL10_XL, tmp);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL2_G, gyr_scale);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL2_G, gyr_scale_125);
+    /* configure gyroscope data ready pin */
+    gpio_init_in(gyr_pin, GPIO_NOPULL);
+    i2c_release(dev->i2c);
+    if (res != 7)
+    {
+        DEBUG("[ FAIL ]\n");
+        return -1;
+    }
+    DEBUG("[ OK ]\n");
+
+    return 0;
+}
+
+int lsm6ds3_read_acc(lsm6ds3_t *dev, lsm6ds3_3d_data_t *data)
+{
+    int res;
+    char tmp;
+
+    i2c_acquire(dev->i2c);
+    i2c_read_reg(dev->12c, dev-acc_address, LSM6DS3_STATUS_REG, &tmp);
+    DEBUG("LSM6DS3: status ... %x\n", tmp);
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL3_C, LSM6DS3_CTRL3_C_BDU);
+    DEBUG("LSM6DS3: retrieve accelerometer values ... ");
+    res = i2c_read_reg(dev->i2c, dev->acc_address,
+                       LSM6DS3_REG_OUT_X_L_XL, &tmp);
+    data->x_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_REG_OUT_X_H_XL, &tmp);
+    data->x_axis |= tmp<<8;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_REG_OUT_Y_L_XL, &tmp);
+    data->y_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_REG_OUT_Y_H_XL, &tmp);
+    data->y_axis |= tmp<<8;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_REG_OUT_Z_L_XL, &tmp);
+    data->z_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_REG_OUT_Z_H_XL, &tmp);
+    data->z_axis |= tmp<<8;
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL3_C, ~(1<<7));
+    i2c_release(dev->i2c);
+
+    data->x_axis = data->x_axis>>4;
+    data->y_axis = data->y_axis>>4;
+    data->z_axis = data->z_axis>>4;
+
+    if (res != 6)
```

Comment:
Except the `+=` or `=` discussion I see eight i2c transfers above. Here you check for six. Did I miss something?